sources/engine/Stride.Physics/Simulation.cs

```diff
@@ -1045,8 +1057,25 @@ internal void Simulate(float deltaTime)
 
             SimulationProfiler = Profiler.Begin(PhysicsProfilingKeys.SimulationProfilingKey);
 
-            if (discreteDynamicsWorld != null) discreteDynamicsWorld.StepSimulation(deltaTime, MaxSubSteps, FixedTimeStep);
-            else collisionWorld.PerformDiscreteCollisionDetection();
+            if (discreteDynamicsWorld != null)
+            {
+                carriedDelta += deltaTime;
+                for (; carriedDelta > FixedTimeStep; )
+                {
+                    var ts = Stopwatch.GetTimestamp();
```

Comment:
If I understand things correctly this should be outside the loop, to limit the entire update duration, not each substep

Reply:
Oof, good catch, thanks !